### PR TITLE
fix: Monitoring Correctness

### PR DESCRIPTION
**Pull request by NisargIO**
1. performance
2. data correctness
3. still waiting on interaction metadata to fix interactions being negative / too high

### CODE REVIEW

File: packages/scan/src/core/monitor/index.ts [hunk 1]
```diff
@@ -149,44 +146,41 @@ const aggregateComponentRenderToInteraction = (
...
   const monitor = Store.monitor.value;
-  if (monitor && monitor.interactions && monitor.interactions.length > 0) {
-    const latestInteraction =
-      monitor.interactions[monitor.interactions.length - 1];
-
-    let totalTime = 0;
-    for (const render of renders) {
-      totalTime += render.time ?? 0;
-    }
+  if (!monitor || !monitor.interactions || monitor.interactions.length === 0)
```
Comment (RobPruzan) [inline packages/scan/src/core/monitor/index.ts:149]: don't have to make this change, but for weird stuff like this in the future just comment 

`// invariant: monitor must exist when calling this function`

one of the few type of comments i find useful and makes future cleanups easier

Comment (NisargIO) [reply packages/scan/src/core/monitor/index.ts:149]: That conditional was there before, just made it a early return

File: packages/scan/src/core/monitor/index.ts [hunk 2]
```diff
@@ -149,44 +146,41 @@ const aggregateComponentRenderToInteraction = (
...
+  component.renders += rendersCount;
 
-    component.renders += renders.length;
-    if (!component.totalTime) {
-      component.totalTime = 0;
-    }
-    component.totalTime += component.totalTime
-      ? component.totalTime + totalTime
-      : totalTime;
+  // We leave the times undefined to differentiate between a 0ms render and a non-profiled render.
```
Comment (RobPruzan) [inline packages/scan/src/core/monitor/index.ts:178]: doesn't the json replacer kill properties with undefined values?

**Review by RobPruzan — APPROVED** [review]
lgtm, merge when ready and ill test everything e2e before a release

File: packages/scan/src/core/monitor/index.ts (hunk 1 above)
Comment (RobPruzan) [reply packages/scan/src/core/monitor/index.ts:149]: yeah not trying to nit, just thought was a good opportunity to say that

File: packages/scan/src/core/monitor/index.ts (hunk 2 above)
Comment (NisargIO) [reply packages/scan/src/core/monitor/index.ts:178]: Yes but in clock house it can be nullable

**Pull request merged**